### ISSUE TITLE
[FEATURE]-Add generic ReceiveQueue and TranmitQueue classes

### ISSUE THREAD

**Issue by ndorin** [PepperDash/PepperDashCore#53]
**Is your feature request related to a problem? Please describe.**
Receive and transmit queueing are tricky.  If we had a generic class for queuing messages of either type `string` or `byte[]` it would make life MUCH easier.

**Describe the solution you'd like**
Create classes like those created to fix the thread abort issue in the MC plugin to be used generically.

**Comment by ngenovese11** [comment]
I'm realizing the Core is a sharp library designed to work with simpl+, as such I don't have access to Crestron.Threading.  I won't be able to implement the same pattern as MC.

**Comment by ndorin** [comment]
Good point Nick.  At very least we should put this in PepperDash.Essentials.Core then so it’s available to all plugins.  And likely create interfaces.


Thanks,

Neil Dorin
Software Developer /SC
PepperDash
www.pepperdash.com<http://www.pepperdash.com/>


PepperDash Mobile Control<http://www2.pepperdash.com/learn-about-mobile-control>™ - BYOD Crestron Control, no special app required

**Comment by ngenovese11** [comment]
Yes I think that Essentials.Core would be the place for it
